gulpfile.ts

```diff
@@ -22,14 +24,33 @@ interface APIModel extends GithubObject {
   outputPath: string
 }
 
+// configure dotenv and get github token
+dotenv.config()
+const GITHUB_TOKEN = env.get('GITHUB_TOKEN').required().asString()
+
 // Using authentication to increases Github API rate limit.
-const octokit = new Octokit()
+const octokit = new Octokit({ auth: GITHUB_TOKEN })
 const OWNER = 'amzn'
 const REPO = 'selling-partner-api-models'
 const REDUNDANT_FILES: string[] = ['.gitignore', '.openapi-generator-ignore', 'git_push.sh']
 const REDUNDANT_DIRECTORIES: string[] = ['.openapi-generator']
 const EXCLUDE_EXPORTED_OBJECTS = new Set(['ErrorList', 'Error'])
 
+async function hasNewCommitsSinceYesterday(repoPath = 'models'): Promise<boolean> {
+  const date = new Date(new Date().toISOString())
+  date.setUTCHours(-24, 0, 0, 0)
```

Comment:
This should be configurable through env too, and have a default. We can just set the hour frequency.

That way we can change that in one place only. E.g. if we decide to run it weekly, all we need to change is the GH Action code.